apiserver/pkg/http/client.go

```diff
@@ -78,13 +83,13 @@ func (krc *KuberayAPIServerClient) CreateComputeTemplate(request *api.CreateComp
 	httpRequest.Header.Add("Accept", "application/json")
 	httpRequest.Header.Add("Content-Type", "application/json")
 
-	bodyBytes, status, err := krc.executeRequest(httpRequest, createURL)
+	bodyBytes, status, err := krc.executeHttpRequest(httpRequest, createURL)
 	if err != nil {
 		return nil, status, err
 	}
 	computeTemplate := &api.ComputeTemplate{}
 	if err := krc.unmarshaler.Unmarshal(bodyBytes, computeTemplate); err != nil {
-		return nil, status, nil
+		return nil, status, fmt.Errorf("failed to unmarshal: %+w", err)
```

Comment:
Would you mind explaining the behavior change from the user's perspective, comparing the old behavior (nil error) with the current behavior (non-nil error)?

Reply:
Unmarshal error gets propagated